config/config.go

```diff
@@ -1,34 +1,48 @@
 package config
 
 import (
+	"fmt"
 	"io/ioutil"
-	"github.com/hashicorp/hcl"
-	"github.com/gruntwork-io/terragrunt/dynamodb"
-	"github.com/gruntwork-io/terragrunt/remote"
+
 	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/locks"
+	"github.com/gruntwork-io/terragrunt/remote"
+	"github.com/hashicorp/hcl"
 )
 
-const TERRAGRUNT_CONFIG_FILE = ".terragrunt"
+const configFilePath = ".terragrunt"
+
+// Config represents a parsed and expanded configuration
+type Config struct {
+	Lock        locks.Lock
+	RemoteState *remote.RemoteState
+}
 
-// A common interface with all fields that could be in the .terragrunt config file.
-type TerragruntConfig struct {
-	DynamoDbLock *dynamodb.DynamoDbLock
-	RemoteState  *remote.RemoteState
+// fileConfig represents the configuration supported in the .terragrunt file
+type fileConfig struct {
```

Comment:
`terragruntFileConfig` or even `terragruntConfigFile` may be a slightly clearer name here.